File: src/main/java/com/timepath/hl2/io/demo/MessageType.java
package com.timepath.hl2.io.demo;

import org.jetbrains.annotations.Nullable;

/**
 * @author TimePath
 */
public enum MessageType {
    Signon(1),
    Packet(2),
    Synctick(3),
    ConsoleCmd(4),
    UserCmd(5),
    DataTables(6),
    Stop(7),
    StringTables(8);

    MessageType(int i) {
    }

    @Nullable
    static MessageType get(int i) {
        MessageType[] vals = MessageType.values();
        if ((i < 1) || (i > vals.length)) {
            return null;
        }
        return vals[i - 1];
    }
}

File: src/main/kotlin/com/timepath/hl2/io/demo/MessageType.kt
package com.timepath.hl2.io.demo

public enum class MessageType(private val id: Int) {

    Signon(1),
    Packet(2),
    Synctick(3),
    ConsoleCmd(4),
    UserCmd(5),
    DataTables(6),
    Stop(7),
    StringTables(8);

    companion object {
        fun get(i: Int): MessageType? = values().first { it.id == i }
    }
}
